registry/purview-registry/registry/purview_registry.py

```diff
@@ -568,17 +571,47 @@ def _register_feathr_feature_types(self):
     def _upload_entity_batch(self, entity_batch:list[AtlasEntity]):
         # we only support entity creation, update is not supported. 
         # setting lastModifiedTS ==0 will ensure this, if another entity with ts>=1 exist
-        # upload funtion will fail with 412 Precondition fail.
+        # upload function will fail with 412 Precondition fail.
         for entity in entity_batch:
-            entity.lastModifiedTS="0"
-            results = self.purview_client.upload_entities(
-                batch=entity)
-            if results:
-                dict = {x.guid: x for x in entity_batch}
-                for k, v in results['guidAssignments'].items():
-                    dict[k].guid = v
+            self._upload_single_entity(entity)
+    
+    def _upload_single_entity(self, entity:AtlasEntity):
+        try:
+            """
+            Try to find existing entity/process first, if found, return the existing entity's GUID
+            """
+            id = self.get_entity_id(entity.qualifiedName)
+            response =  self.purview_client.get_entity(id)['entities'][0]
+            j = entity.to_json()
+            if j["typeName"] == response["typeName"]:
+                if j["typeName"] == "Process":
+                    if response["attributes"]["qualifiedName"] != j["attributes"]["qualifiedName"]:
+                        raise RuntimeError("The requested entity conflicts with the existing entity in PurView")
+                else:
+                    if "type" in response['attributes'] and response["typeName"] in ("feathr_anchor_feature_v1", "feathr_derived_feature_v1"):
+                        conf = ConfigFactory.parse_string(response['attributes']['type'])
+                        response['attributes']['type'] = dict(conf)
+                    keys = set([to_snake(key) for key in j["attributes"].keys()]) - set(["qualified_name"])
+                    keys.add("qualifiedName")
+                    for k in keys:
+                        if response["attributes"][k] != j["attributes"][k]:
+                            raise RuntimeError("The requested entity conflicts with the existing entity in PurView")
+                entity.guid = response["guid"]
+                return
             else:
-                raise RuntimeError("Feature registration failed.", results)            
+                raise RuntimeError("The requested entity conflicts with the existing entity in PurView")
+        except AtlasException as e:
+            pass
```

Comment:
Should we print out the exception rather than simply pass?

Reply:
This `pass` is deliberated, Atlas will throw an exception when the entity is not found, this code piece uses this behavior to check entity existence.